src/int.rs

```diff
@@ -344,10 +344,11 @@ impl Int {
     /// # use js_int::{int, Int};
     /// assert_eq!(int!(100).saturating_add(int!(1)), int!(101));
     /// assert_eq!(Int::MAX.saturating_add(int!(1)), Int::MAX);
+    /// assert_eq!(Int::MIN.saturating_add(int!(-1)), Int::MIN);
     /// ```
     #[must_use]
     pub fn saturating_add(self, rhs: Self) -> Self {
-        self.checked_add(rhs).unwrap_or(Self::MAX)
+        self.checked_add(rhs).unwrap_or(if self.0 > -rhs.0 { Self::MAX } else { Self::MIN })
```

Comment:
This should be an `unwrap_or_else` since the comparison should only be done if `checked_add` returns `None`. I'm also wondering whether comparison to `0` might be more efficient (if `self` is negative it can only saturate at `Self::MIN`, if it's positive it can only saturate at `Self::MAX`). Maybe even

```rust
match self.cmp(int!(0)) {
    Ordering::Less => Self::MIN,
    Ordering::Equal => unsafe { std::hint::unreachable_unchecked() },
    Ordering::Greater => Self::MAX,
}
```

to give an extra hint to the optimizer? Might be overkill.

Reply:
Oh right, we are adding two already-capped values together – then comparing with 0 works, too.

Regarding performance, there is no way of knowing except for benchmarking ;) Though clippy suggests that ifs are currently faster: https://rust-lang.github.io/rust-clippy/master/#comparison_chain